community-engagement-plan.md

```diff
@@ -0,0 +1,210 @@
+high-level goals are:
+
+- [Operator][]-driven deployment and management of FreeIPA
+  installations (consisting of multiple servers).  The operator will
+  manage topology, scaling, monitoring, upgrades and backup/restore.
+
+- Initially, to support deployment on AWS and GCP.  Support for
+  clusters on other providers (or bare metal) is nice-to-have, and
+  is an opportunity for community collaboration.
+
+- Use non-privileged workloads for secure multi-tenancy.
+
+- Eventually, to enable FreeIPA managed service offerings.
+  
+[Operator]: https://kubernetes.io/docs/concepts/extend-kubernetes/operator/
+
+## Goals
+
+This document outlines a community engagement plan, including:
+
+- Clear communication of our project "why", including use cases.
+
+- Transparent communication about the project goals, roadmap, status
+  and challenges.
+
+- Raise awareness of the project within the broader FreeIPA,
+  OpenShift and open source identity management communities.
+
+- Establish communication channels for interested people to engage
+  with the project and its developers.
+
+- Encourage community contributions in various forms including
+  testing, bug reports, code, documentation and ideas.
+
+
+## Implementation
+
+### Project name
+
+Internally, the project is called "IDMOCP".  We have to decide if
+this name is suitable for the public face of the project, or whether
+something else, like *FreeIPA on OpenShift* is better.
+
+**DECIDE**
+
+- IDMOCP?
+- Something else? (please share your ideas)
+
+
+### Code of Conduct
+
+The project should have a code of conduct, and we should make it
+clear what that is, and violation will be dealt with.  We may be
+able to operate under the FreeIPA code of conduct.
+
+**NOTE** I have follow up with FreeIPA project about operating under
+their CoC and reporting/enforcement mechanisms.
+
+### Communication channels
+
+#### IRC
+
+FreeIPA has an established IRC channel (`#freeipa` on
+`libera.chat`), that receives a fair bit of community engagement.
+
+**DECIDE** do we:
+
+- use `#freeipa` for IDMOCP discussion?  (We should consult FreeIPA
+  project about this option)
+- create our own IRC channel, e.g. `#freeipa-openshift`
+- not use IRC as a primary channel, and redirect questions in 
+  `#freeipa` to another venue.
+
+
+#### Slack
+
+Slack seems popular in the OpenShift and Kubernetes communities.  We
+could create a Slack channel specifically for the IDMOCP project.
+Slack has the advantage of keeping a permanent, searchable record of
+messages exchanged, and is a more expressive medium than IRC (links,
+attachments, images, etc).
+
+**DECIDE**
+
+- yes to slack
+- no to slack, but yes to something similar
+- no to slack or similar 
+
+#### Mailing list
+
+**DECIDE**
+
+In light of other communication channels (including GitHub issues or
+other ticket systems), do we require a public mailing list for the
+IDMOCP effort?
+
+- yes to mailing list
```

Comment:
If yes, we should also consider using `freeipa-devel@` and `freeipa-users@`, at least in the beginning when there is low traffic.  @pvoborni do you have thoughts about it?

Reply:
Both freeipa-devel and freeipa-users looks good to me. Discussions on users might bring awareness to this project. freeipa-devel is not much used lately and it would be nice if it was. Personally, I'd disable PR notifications which are forwarded to freeipa-devel as they might overshadow everything else if you don't have the right filters set.

Reply:
OK, thanks @pvoborni and @ragevou.  I think it makes sense to use existing FreeIPA channels right now, plus Slack because that is where a lot of CoreOS / OpenShift people/projects are, and consider Telegram also, direct or via bridge.